pyqldbsamples/insert_ion_types.py

```diff
@@ -92,95 +95,135 @@ def delete_table(transaction_executor, table_name):
     return len(list(cursor))
 
 
-TABLE_NAME = 'IonTypes'
+def insert_and_verify_ion_types(session):
+    """
+    Insert all the supported Ion types and Python values that are convertible to Ion into a ledger and verify that they
+    are stored and can be retrieved properly, retaining their original properties.
+
+    :type session: :py:class:`pyqldb.session.base_qldb_session.BaseQldbSession`
+    :param session: A QLDB session object.
+    """
+    python_bytes = str.encode('hello')
+    python_bool = True
+    python_float = float('0.2')
+    python_decimal = Decimal('0.1')
+    python_string = "string"
+    python_int = 1
+    python_null = None
+    python_datetime = datetime(2016, 12, 20, 5, 23, 43)
+    python_list = [1, 2]
+    python_dict = {"brand": "Ford"}
+
+    ion_clob = convert_object_to_ion(loads('{{"This is a CLOB of text."}}'))
+    ion_blob = convert_object_to_ion(python_bytes)
+    ion_bool = convert_object_to_ion(python_bool)
+    ion_decimal = convert_object_to_ion(python_decimal)
+    ion_float = convert_object_to_ion(python_float)
+    ion_int = convert_object_to_ion(python_int)
+    ion_list = convert_object_to_ion(python_list)
+    ion_null = convert_object_to_ion(python_null)
+    ion_sexp = convert_object_to_ion(loads('(cons 1 2)'))
+    ion_string = convert_object_to_ion(python_string)
+    ion_struct = convert_object_to_ion(python_dict)
+    ion_symbol = convert_object_to_ion(SymbolToken(text='abc', sid=123))
+    ion_timestamp = convert_object_to_ion(python_datetime)
+
+    ion_null_clob = convert_object_to_ion(loads('null.clob'))
+    ion_null_blob = convert_object_to_ion(loads('null.blob'))
+    ion_null_bool = convert_object_to_ion(loads('null.bool'))
+    ion_null_decimal = convert_object_to_ion(loads('null.decimal'))
+    ion_null_float = convert_object_to_ion(loads('null.float'))
+    ion_null_int = convert_object_to_ion(loads('null.int'))
+    ion_null_list = convert_object_to_ion(loads('null.list'))
+    ion_null_sexp = convert_object_to_ion(loads('null.sexp'))
+    ion_null_string = convert_object_to_ion(loads('null.string'))
+    ion_null_struct = convert_object_to_ion(loads('null.struct'))
+    ion_null_symbol = convert_object_to_ion(loads('null.symbol'))
+    ion_null_timestamp = convert_object_to_ion(loads('null.timestamp'))
+
+    session.execute_lambda(lambda transaction_executor: create_table(transaction_executor, TABLE_NAME)
+                           and insert_documents(transaction_executor, TABLE_NAME, [{'Name': 'val'}])
```

Comment:
let's take note for the next version about not over engineering this sample code.

Reply:
Agreed, should be as simple as possible.